### PR TITLE
FRONT-826: Upgrade streamr-client version to 6.0.0-alpha25 (OCR enabled)

### PR DESCRIPTION
**Pull request by bearni95**
Get the chat working with the Polygon mainnet contracts and their new methods

### CODE REVIEW

File: src/components/pages/Chat/MessageTransmitter.tsx [hunk 1]
```diff
@@ -70,7 +72,13 @@ export default function MessageTransmitter({ children }: Props) {
                     }
 
                     await (async () => {
-                        const stream = await streamrClient.getStream(roomId)
+                        const metamaskStreamrClient = new StreamrClient({
+                            auth: { ethereum: ethereumProvider as any },
+                        })
```
Comment (mondoreale) [inline src/components/pages/Chat/MessageTransmitter.tsx:77]: This MM streamrClient is used often. Maybe we should throw it into the global, state alongside `ethereumProvider`?

File: src/hooks/useCreateRoom.ts [hunk 2]
```diff
@@ -28,23 +30,32 @@ export default function useCreateRoom(): () => Promise<void> {
...
 
-        const stream = await streamrClient.createStream({
+        const metamaskStreamrClient = new StreamrClient({
+            auth: { ethereum: ethereumProvider as any },
+        })
+        const stream = await metamaskStreamrClient.createStream({
             id,
             partitions: 2,
-            requireEncryptedData: false,
+            // requireEncryptedData: false,
```
Comment (mondoreale) [inline src/hooks/useCreateRoom.ts:41]: Just drop it.

**Review by mondoreale — APPROVED** [review]
Very nice. I created a low-prio ticket – [`FRONT-839`](https://linear.app/streamr/issue/FRONT-839/wrong-network-warning).

File: src/components/pages/Chat/MessageTransmitter.tsx (hunk 1 above)
Comment (bearni95) [reply src/components/pages/Chat/MessageTransmitter.tsx:77]: agreed, i'll replicate `ethereumProvider`'s schema with the `metamaskStreamrClient`

File: src/components/Store.tsx [hunk 3]
```diff
@@ -115,12 +115,16 @@ function reducer(state: ChatState, action: A): ChatState {
                       },
                   }
         case ActionType.Reset:
+            const client = new StreamrClient({
+                auth: { ethereum: state.ethereumProvider as any },
+            })
```
Comment (mondoreale) [inline src/components/Store.tsx:120]: Wrong spot. Let's do it in `ActionType.SetEthereumProvider` and here let's just do
```ts
return {
    ...getInitialChatState(),
    // Keep the provider. We may wanna extract it from "chat" state. It's actually more
    // of a browser/app state piece.
    ethereumProvider: state.ethereumProvider,
    ethereumProviderReady: state.ethereumProviderReady,
    metamaskStreamrClient: state.metamaskStreamrClient, // HERE
}
```

Comment (mondoreale) [reply src/components/Store.tsx:120]: This would create a metamask client on logout. ;)

File: src/getters/getInitialStoreState.ts [hunk 4]
```diff
@@ -5,6 +5,7 @@ export default function getInitialChatState(): ChatState {
         account: undefined,
         drafts: {},
         ethereumProvider: undefined,
+        metamaskStreamrClient: undefined,
```
Comment (mondoreale) [inline src/getters/getInitialStoreState.ts:8]: 👍🏻

**Review by mondoreale — CHANGES_REQUESTED** [review]

**Review by mondoreale — APPROVED** [review]

**Pull request merged**